### PR TITLE
Fix some bugs running Export Glyph Textures

### PR DESCRIPTION
**Pull request by arcturus2**
### Description of the Change

I looked at this bit of code during my review of #446 and couldn't help but fix the following:
* Fixed some exceptions seen running Experimental -> Glyph Textures
* Removed some comments no longer used
* Added some finals

### CODE REVIEW

File: CoreOpenGLDisplay/src/au/gov/asd/tac/constellation/visual/opengl/utilities/SharedDrawable.java [hunk 1]
```diff
@@ -129,28 +122,39 @@ private static void init() {
...
+                final int extensionStart = baseFileName.lastIndexOf('.');
+                if (extensionStart > 0) {
+                    baseFileName = baseFileName.substring(0, extensionStart);
                 }
-            }
-        });
+
+                for (int page = 0; page < glyphManager.getGlyphPageCount(); page++) {
+                    final File outputFile = new File(baseFileName + SeparatorConstants.UNDERSCORE + page + ".png");
+                    try ( OutputStream out = new FileOutputStream(outputFile)) {
```
Comment (cygnus-x-1) [inline CoreOpenGLDisplay/src/au/gov/asd/tac/constellation/visual/opengl/utilities/SharedDrawable.java:148]: out can be final, also remove the rogue space.

**Comment by formalhaut69** [comment]
@arcturus2 I've tested the exportGlyphTextures method from this branch and this image is the output that I get. I'm not really sure what this method is meant to do but if this output is correct then I'm happy with these changes.

image

File: CoreOpenGLDisplay/src/au/gov/asd/tac/constellation/visual/opengl/utilities/SharedDrawable.java [hunk 2]
```diff
@@ -129,28 +122,39 @@ private static void init() {
...
+                        glyphManager.writeGlyphBuffer(page, out);
+                    } catch (IOException ex) {
+                        // Nothing to do here - this is just for developer testing.
```
Comment (antares1470) [inline CoreOpenGLDisplay/src/au/gov/asd/tac/constellation/visual/opengl/utilities/SharedDrawable.java:151]: is the intention to remove this once developer testing is done?

**Comment by cygnus-x-1** [comment]
> @arcturus2 I've tested the exportGlyphTextures method from this branch and this image is the output that I get. I'm not really sure what this method is meant to do but if this output is correct then I'm happy with these changes.
> 
> image

That looks correct - the function should create an image representing the texture map of glyphs sent to the graphics card which it then uses to render labels in a graph. It looks like that because it uses an algorithm to tightly pack the characters in order to fit as much as possible into a single texture map, to reduce data transfer to/from the graphics card. @algol60 will be able to describe this better than me.

**Comment by formalhaut69** [comment]
> 
> 
> > @arcturus2 I've tested the exportGlyphTextures method from this branch and this image is the output that I get. I'm not really sure what this method is meant to do but if this output is correct then I'm happy with these changes.
> > image
> 
> That looks correct - the function should create an image representing the texture map of glyphs sent to the graphics card which it then uses to render labels in a graph. It looks like that because it uses an algorithm to tightly pack the characters in order to fit as much as possible into a single texture map, to reduce data transfer to/from the graphics card. @algol60 will be able to describe this better than me.

Cool, thanks for the explanation @cygnus-x-1

File: CoreOpenGLDisplay/src/au/gov/asd/tac/constellation/visual/opengl/utilities/SharedDrawable.java (hunk 1 above)
Comment (arcturus2) [reply CoreOpenGLDisplay/src/au/gov/asd/tac/constellation/visual/opengl/utilities/SharedDrawable.java]: Good pickup regarding the `final` and rogue space. I added the `final` but the space seems to be part of the auto-formatter. I'd say that's a NB 11.3 bug.

File: CoreOpenGLDisplay/src/au/gov/asd/tac/constellation/visual/opengl/utilities/SharedDrawable.java (hunk 2 above)
Comment (arcturus2) [reply CoreOpenGLDisplay/src/au/gov/asd/tac/constellation/visual/opengl/utilities/SharedDrawable.java]: I'd like to keep this plugin around - so I've replaced the comment with a `logger` call.

**Comment by arcturus2** [comment]
@cygnus-x-1 I'm going to merge this one given I believe I've addressed the minor feedback, cheers.

**Pull request merged**